File: DSA/akashyadav_590014126/Day_82_Question_2.py
class TreeNode:
    def __init__(self, val):
        self.val = val
        self.left = None
        self.right = None

def isBalanced(root):
    def check(node):
        if not node:
            return 0  
        left = check(node.left)
        if left == -1:
            return -1  
        right = check(node.right)
        if right == -1:
            return -1  
        if abs(left - right) > 1:
            return -1  
        return max(left, right) + 1  

    return check(root) != -1

File: DSA/VenikaPanwar_590013755/Day_82_Question_2.java
public class Day_82_Question_2 {

    static class TreeNode {
        int val;
        TreeNode left, right;
        TreeNode(int val) { this.val = val; }
    }

    // Returns -1 if not balanced, else returns height
    private static int checkHeight(TreeNode root) {
        if (root == null) return 0;
        int left = checkHeight(root.left);
        int right = checkHeight(root.right);
        if (left == -1 || right == -1 || Math.abs(left - right) > 1) return -1;
        return Math.max(left, right) + 1;
    }

    public static boolean isBalanced(TreeNode root) {
        return checkHeight(root) != -1;
    }

    public static void main(String[] args) {
        // Balanced tree
        TreeNode root1 = new TreeNode(4);
        root1.left = new TreeNode(2);
        root1.right = new TreeNode(6);
        root1.left.left = new TreeNode(1);
        root1.left.right = new TreeNode(3);
        root1.right.left = new TreeNode(5);
        root1.right.right = new TreeNode(7);
        System.out.println("Is balanced (Example 1): " + (isBalanced(root1) ? "Yes" : "No")); // Output: Yes

        // Unbalanced tree
        TreeNode root2 = new TreeNode(1);
        root2.right = new TreeNode(2);
        root2.right.right = new TreeNode(3);
        System.out.println("Is balanced (Example 2): " + (isBalanced(root2) ? "Yes" : "No")); // Output: No
    }
}
